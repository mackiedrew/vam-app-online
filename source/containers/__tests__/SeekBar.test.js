import {
  SeekBar as Subject,
  mapStateToProps,
  mapDispatchToProps
} from "../SeekBar";

const mockProps = {
  seekPosition: 10,
  currentlyPlaying: true
};

describe("<SeekBar />", () => {
  it("renders without crashing", () => {
    const subject = shallow(<Subject {...mockProps} />);
    expect(subject.is("div.seek-bar")).toBe(true);
  });

  it("seekSeconds call appropriate function with no sampleRate", () => {
    const mockShiftSeekPosition = sinon.spy();
    const subject = shallow(
      <Subject {...mockProps} shiftSeekPosition={mockShiftSeekPosition} />
    );
    subject.instance().seekSeconds(10);
    expect(mockShiftSeekPosition.called).toBe(true);
  });

  it("seekSeconds call appropriate function with a sampleRate", () => {
    const mockShiftSeekPosition = sinon.spy();
    const subject = shallow(
      <Subject {...mockProps} shiftSeekPosition={mockShiftSeekPosition} />
    );
    subject.instance().seekSeconds(10, 33333);
    expect(mockShiftSeekPosition.called).toBe(true);
  });

  it("handlePlus1 call appropriate function", () => {
    const mockShiftSeekPosition = sinon.spy();
    const subject = shallow(
      <Subject {...mockProps} shiftSeekPosition={mockShiftSeekPosition} />
    );
    subject.instance().handlePlus1();
    expect(mockShiftSeekPosition.called).toBe(true);
  });

  it("handleMinus1 call appropriate function", () => {
    const mockShiftSeekPosition = sinon.spy();
    const subject = shallow(
      <Subject {...mockProps} shiftSeekPosition={mockShiftSeekPosition} />
    );
    subject.instance().handleMinus1();
    expect(mockShiftSeekPosition.called).toBe(true);
  });

  it("handleViewNext call appropriate function", () => {
    const mockShiftView = sinon.spy();
    const subject = shallow(
      <Subject {...mockProps} shiftView={mockShiftView} />
    );
    subject.instance().handleViewNext();
    expect(mockShiftView.called).toBe(true);
  });

  it("handleViewPrevious call appropriate function", () => {
    const mockShiftView = sinon.spy();
    const subject = shallow(
      <Subject {...mockProps} shiftView={mockShiftView} />
    );
    subject.instance().handleViewPrevious();
    expect(mockShiftView.called).toBe(true);
  });

  it("handleZoomIn call appropriate function", () => {
    const mockZoomIn = sinon.spy();
    const subject = shallow(<Subject {...mockProps} zoomIn={mockZoomIn} />);
    subject.instance().handleZoomIn();
    expect(mockZoomIn.called).toBe(true);
  });

  it("handleZoomOut call appropriate function", () => {
    const mockZoomOut = sinon.spy();
    const subject = shallow(<Subject {...mockProps} zoomOut={mockZoomOut} />);
    subject.instance().handleZoomOut();
    expect(mockZoomOut.called).toBe(true);
  });

  it("handleTogglePlay call appropriate function", () => {
    const mockToggleCurrentlyPlaying = sinon.spy();
    const subject = shallow(
      <Subject
        {...mockProps}
        toggleCurrentlyPlaying={mockToggleCurrentlyPlaying}
      />
    );
    subject.instance().handleTogglePlay();
    expect(mockToggleCurrentlyPlaying.called).toBe(true);
  });

  it("renders correctly", () => {
    const tree = renderer.create(<Subject {...mockProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  describe("mapStateToProps()", () => {
    const mockState = {
      tracks: {
        currentlyPlaying: 1,
        seekPosition: 2
      }
    };

    it("returns object with expected properties", () => {
      const { currentlyPlaying, seekPosition } = mapStateToProps(mockState);
      expect(currentlyPlaying).toBe(1);
      expect(seekPosition).toBe(2);
    });
  });

  describe("mapDispatchToProps()", () => {
    it("returns object with expected properties", () => {
      const mockDispatch = sinon.spy();
      const {
        zoomIn,
        zoomOut,
        shiftView,
        shiftSeekPosition,
        toggleCurrentlyPlaying
      } = mapDispatchToProps(mockDispatch);
      expect(typeof zoomIn).toBe("function");
      expect(typeof zoomOut).toBe("function");
      expect(typeof shiftView).toBe("function");
      expect(typeof shiftSeekPosition).toBe("function");
      expect(typeof toggleCurrentlyPlaying).toBe("function");
    });
  });
});
