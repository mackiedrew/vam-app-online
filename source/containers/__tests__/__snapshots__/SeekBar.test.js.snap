// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SeekBar /> renders correctly 1`] = `
<div
  className="seek-bar"
>
  <div
    className="seek-controls"
  >
    <button
      className="seek-minus-1"
      onClick={[Function]}
    >
      <i
        className="material-icons icon"
        style={
          Object {
            "fontSize": "24px",
            "height": "24px",
            "width": "24px",
          }
        }
      >
        skip_previous
      </i>
    </button>
    <button
      className="toggle-button"
      onClick={[Function]}
    >
      <i
        className="material-icons icon"
        style={
          Object {
            "fontSize": "24px",
            "height": "24px",
            "width": "24px",
          }
        }
      >
        pause
      </i>
    </button>
    <button
      className="seek-plus-1"
      onClick={[Function]}
    >
      <i
        className="material-icons icon"
        style={
          Object {
            "fontSize": "24px",
            "height": "24px",
            "width": "24px",
          }
        }
      >
        skip_next
      </i>
    </button>
  </div>
  <div
    className="view-controls"
  >
    <button
      className="view-previous"
      onClick={[Function]}
    >
      <i
        className="material-icons icon"
        style={
          Object {
            "fontSize": "24px",
            "height": "24px",
            "width": "24px",
          }
        }
      >
        navigate_before
      </i>
    </button>
    <button
      className="view-next"
      onClick={[Function]}
    >
      <i
        className="material-icons icon"
        style={
          Object {
            "fontSize": "24px",
            "height": "24px",
            "width": "24px",
          }
        }
      >
        navigate_next
      </i>
    </button>
    <button
      className="zoom-in"
      onClick={[Function]}
    >
      <i
        className="material-icons icon"
        style={
          Object {
            "fontSize": "24px",
            "height": "24px",
            "width": "24px",
          }
        }
      >
        zoom_in
      </i>
    </button>
    <button
      className="zoom-out"
      onClick={[Function]}
    >
      <i
        className="material-icons icon"
        style={
          Object {
            "fontSize": "24px",
            "height": "24px",
            "width": "24px",
          }
        }
      >
        zoom_out
      </i>
    </button>
    <div
      className="current-time"
    >
      00:00:00:000
    </div>
  </div>
</div>
`;
